File: src/test/java/com/dotnomi/fabricdependencyinjection/testclasses/circular/CircularB.java
package com.dotnomi.fabricdependencyinjection.testclasses.circular;

import com.dotnomi.fabricdependencyinjection.annotation.ModInject;
import com.dotnomi.fabricdependencyinjection.annotation.ModScoped;

@ModScoped
@SuppressWarnings("unused")
public class CircularB {
  @ModInject
  public CircularB(CircularA circularA) {}
}

File: src/test/java/com/dotnomi/fabricdependencyinjection/kotlin/testclasses/circular/CircularB.kt
package com.dotnomi.fabricdependencyinjection.kotlin.testclasses.circular

import com.dotnomi.fabricdependencyinjection.annotation.ModInject
import com.dotnomi.fabricdependencyinjection.annotation.ModScoped

@ModScoped
class CircularB @ModInject constructor(circularA: CircularA)
